From cc3e1abaa1e7ad85c6a299fe30a36e0cb732e1c8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 15 Dec 2022 14:51:20 -0800
Subject: [PATCH] compat: Define off64_t as off_t on linux

This helps compile with latest musl, where off64_t is not defined
unless _LARGEFILE64_SOURCE is defined, on glibc _LARGEFILE64_SOURCE is
defined if _GNU_SOURCE is defined, so the problem is only seen with musl

Upstream-Status: Submitted [https://github.com/lttng/lttng-tools/pull/164]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/common/compat/fcntl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/compat/fcntl.h b/src/common/compat/fcntl.h
index 5e566e50..d19a6ece 100644
--- a/src/common/compat/fcntl.h
+++ b/src/common/compat/fcntl.h
@@ -7,7 +7,6 @@
 
 #ifndef _COMPAT_FCNTL_H
 #define _COMPAT_FCNTL_H
-
 #include <fcntl.h>
 #include <sys/types.h>
 
@@ -22,6 +21,7 @@ typedef off64_t loff_t;
 #endif
 
 #ifdef __linux__
+typedef off_t off64_t;
 extern int compat_sync_file_range(int fd, off64_t offset, off64_t nbytes,
 		unsigned int flags);
 #define lttng_sync_file_range(fd, offset, nbytes, flags) \
